### ISSUE TITLE
Topbar dropdown not working with Turbolinks

### ISSUE THREAD

**Issue by collimarco** [zurb/foundation#6438]
I use Rails 4.2.0 and foundation-rails 5.5.1.0.

I have a dropdown menu in the topbar like this:

```
<nav role="navigation" data-topbar="" class="top-bar">
    <section class="top-bar-section">
        <ul class="right">
          <li class="has-dropdown">
            <a href="/profile">Tester</a>
            <ul class="dropdown">
              <li><a href="/profile/edit">Edit profile</a></li>
              <li><a href="/users/edit">Edit account</a></li>
              <li><a href="/users/sign_out" data-method="delete" rel="nofollow">Log out</a></li>
            </ul>
          </li>
        </ul>
      </section>
</nav>
```

It works as expected until I do the following steps:

1. click "Tester" (the heading of the dropdown menu, notice that it is a link, not a `href="#"`)
2. hover "Tester" again
3. you would expect the dropdown menu to appear as always, instead only a javascript error is displayed in the console saying that something is undefined

I've solved by using jquery-turbolinks gem, but I think that this is only a patch. 

I think that Foundation should work properly with Turbolinks by default.

**Comment by tipsforthings** [comment]
I noticed I had problems with this also. It's a problem with the way the asset initialiser is set up. I had to add the line configure.server_static_files to false to get it to work.

**Comment by rafibomb** [comment]
We've found that initializing Foundation like this helps with the issue
 `$(function(){ $(document).foundation(); });`

**Comment by embersdev** [comment]
I find this works the best and won't require jquery-turbolinks.

$(document).on('ready page:load', function () {
  $(document).foundation();
});

And this for Equalizer if you use it....

$(document).on('page:change', function () {
  $(document).foundation('equalizer', 'reflow');
});

I have both at the bottom of my Application.js

**Comment by smpetrey** [comment]
I had the same problem and as @tipsforthings said it looks like it's a problem with the assets initializer.
setting the `config.serve_static_assets = false` to false fixed it for me.

**Comment by rafibomb** [comment]
Thanks guys - this serves a great reference for people. We re-worked navigation for the new version which will be out soon and we'll close this here. Thanks!

**Comment by Javix** [comment]
Sorry for coming back to this issue. Any working solution for F5  and Rails 4 ? I can't figure out how to make it work. It works only after reloading the page, if not, the dropdown does nothing at all. thank you.

**Comment by Baytars** [comment]
> We've found that initializing Foundation like this helps with the issue
> `$(function(){ $(document).foundation(); });`
> 
> We're looking for ways to avoid this for Foundation Sites 6. Until then, have any suggestions?

Great! This has long been a snag for me.

**Comment by Baytars** [comment]
> We've found that initializing Foundation like this helps with the issue
> `$(function(){ $(document).foundation(); });`
> 
> We're looking for ways to avoid this for Foundation Sites 6. Until then, have any suggestions?

Not always fix. This has also caused a further problem that cost me 3h to fix. New images of clearing-featured-img class failed to load. $('.clearing-featured-img').length is less than the actual sum. Visually you see no new image was added to the gallery. Somehow annoying.

**Comment by DanielRuf** [comment]
> Not always fix. This has also caused a further problem that cost me 3h to fix. New images of clearing-featured-img class failed to load. $('.clearing-featured-img').length is less than the actual sum. Visually you see no new image was added to the gallery. Somehow annoying.

Hi, please open a new issue with all relevant details so we can help you.
